03_letters/lexual/freq_test.go

```diff
@@ -0,0 +1,91 @@
+package main
+
+import (
+	"bytes"
+	"strconv"
+	"testing"
+)
+
+func TestLetters(t *testing.T) {
+	tests := []struct {
+		input    string
+		expected map[rune]int
+	}{
+		{
+			"abcbcc",
+			map[rune]int{
+				'a': 1,
+				'b': 2,
+				'c': 3,
+			},
+		},
+		{
+			"",
+			map[rune]int{},
+		},
+	}
+	for _, test := range tests {
+		actual := letters(test.input)
+		if len(actual) != len(test.expected) {
```

Comment:
I feel this whole test is too verbose. It should be simple to understand what is being tested. You could just use `reflect.DeepEqual`. I understand your comment about the output being integers instead of characters, but you need to balance that with the readability of the test. If you really still want prettier output, I suggest adding a function that creates a string representation of the map as you want and using that in your error message. That keeps the test simple and moves the map formatting logic elsewhere.

Reply:
simplified